### PR TITLE
ENH: Extra Fields for Generic CSV

### PR DESCRIPTION
**Pull request by navtej**
This patch adds the capability to have extra fields as dict (similar to otx and blueliv parsers). 
If a column name starts with `extra.` , the property of extra will be added as dict to "extra" field in event.

### CODE REVIEW

**Comment by wagner-certat** [comment]
This is also part of #935

File: intelmq/bots/parsers/generic/parser_csv.py [hunk 1]
```diff
@@ -65,11 +66,17 @@ def parse_line(self, row, report):
                     value = self.type_translation[value]
                 elif not hasattr(self.parameters, 'type'):
                     continue
+            elif key.startswith("extra."):
+                _, key_ = key.split('.', 1)
```
Comment (wagner-certat) [inline intelmq/bots/parsers/generic/parser_csv.py]: If `key` is `extra.foo.bar` then `key_` is `foo`

File: intelmq/bots/parsers/generic/parser_csv.py [hunk 2]
```diff
@@ -65,11 +66,17 @@ def parse_line(self, row, report):
...
+                additional[key_] = value
+                continue
             event.add(key, value)
 
         if hasattr(self.parameters, 'type')\
                 and "classification.type" not in event:
             event.add('classification.type', self.parameters.type)
+        if len(additional) > 0:
```
Comment (wagner-certat) [inline intelmq/bots/parsers/generic/parser_csv.py]: `if additional` is sufficient.

File: intelmq/bots/parsers/generic/parser_csv.py (hunk 1 above)
Comment (navtej) [inline intelmq/bots/parsers/generic/parser_csv.py]: No, then key is `foo.bar`

```
>>> "extra.foo.bar".split(".",1)
['extra', 'foo.bar']

```

File: intelmq/bots/parsers/generic/parser_csv.py (hunk 2 above)
Comment (navtej) [inline intelmq/bots/parsers/generic/parser_csv.py]: Not really
```
>>> additional = {}
>>> if additional: print("True")
...
>>>
```

**Comment by navtej** [comment]
I started a review for #935 . With one minor change it does offer the same functionality.

File: intelmq/bots/parsers/generic/parser_csv.py (hunk 1 above)
Comment (wagner-certat) [inline intelmq/bots/parsers/generic/parser_csv.py]: Ah, right. Sorry.

File: intelmq/bots/parsers/generic/parser_csv.py (hunk 2 above)
Comment (navtej) [inline intelmq/bots/parsers/generic/parser_csv.py]: ideed, TIL.

Comment (wagner-certat) [inline intelmq/bots/parsers/generic/parser_csv.py]: Which proves that it is sufficient, isn' it?

`bool({}) == False`

**Comment by wagner-certat** [comment]
> With one minor change it does offer the same functionality.

So is there any functionality in this PR which is not in #935?

**Comment by navtej** [comment]
Yes more like a bug, details are already on the opened review @ #935 . However I noticed i started a review but didnt submit it so it was just sitting there probably and you never saw it.

**Pull request closed without merging**